### PR TITLE
Make carboniter 'enabled' config value name/location better [BA-6372]

### PR DESCRIPTION
**Pull request by mcovarr**
and move some other freeze parameters into the `metadata-freezing` block too

### CODE REVIEW

File: cromwell.example.backends/cromwell.examples.conf [hunk 1]
```diff
@@ -536,14 +528,23 @@ services {
...
-    #       max-interval = 5 minutes,
+    #     # Metadata freezing configuration. All of these entries are optional and default to the values shown below.
+    #     # In particular, the default value of `Inf` for `initial-interval` turns off metadata freezing, so an explict
+    #     # non-`Inf` value would need to be chosen for that parameter to turn metadata freezing on.
+    #     metadata-freezing {
+    #       # How often Cromwell should check for metadata ready for freezing. Set the `initial-interval` value to "Inf"
+    #       # (or leave the default unchanged) to turn off metadata freezing. Both interval parameters must be durations,
+    #       # max must be greater than initial, and multiplier must be a number greater than 1.
+    #       initial-interval = Inf
+    #       max-interval = 5 minutes
```
Comment (cjllanwarne) [inline cromwell.example.backends/cromwell.examples.conf:539]: > max must be greater than initial

`5 minutes` > `Inf`?

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala [hunk 2]
```diff
@@ -29,11 +29,12 @@ class CarboniteWorkerActor(carboniterConfig: HybridCarboniteConfig,
...
 
   val backOff = {
-    val scanConfig = carboniterConfig.freezeScanConfig
+    val freezingConfig = carboniterConfig.freezingConfig
     SimpleExponentialBackoff(
-      initialInterval = scanConfig.initialInterval,
-      maxInterval = scanConfig.maxInterval,
-      multiplier = scanConfig.multiplier,
+      // This actor is only created if the initial interval is finite.
+      initialInterval = freezingConfig.initialInterval.asInstanceOf[FiniteDuration],
```
Comment (cjllanwarne) [inline hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala:35]: This assumption makes me a little nervous. 

If this is always true, can we bake the knowledge into the type system? Could we pass it in as a parameter? Or make the `carboniterConfig` a validated subtype of `HybridCarboniteConfig` (`FreezeEnabledHybridCarboniteConfig`, maybe?)

File: hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/HybridCarboniteConfigSpec.scala [hunk 3]
```diff
@@ -145,44 +150,46 @@ class HybridCarboniteConfigSpec extends TestKitSuite("HybridCarboniteConfigSpec"
...
       case Left(e) =>
         e.size shouldBe 3
-        e shouldEqual NonEmptyList.of(
-          "Failed to parse Carboniter 'freeze-scan' stanza (reason 1 of 3): String: 10: Invalid value at 'initial-interval': No number in duration value 'I'",
-          "Failed to parse Carboniter 'freeze-scan' stanza (reason 2 of 3): String: 11: Invalid value at 'max-interval': No number in duration value 'like'",
-          "Failed to parse Carboniter 'freeze-scan' stanza (reason 3 of 3): String: 12: multiplier has type STRING rather than NUMBER"
+        e.toList.toSet shouldEqual Set(
+          "Failed to parse Carboniter 'metadata-freezing' stanza (reason 1 of 3): format error I",
+          "Failed to parse Carboniter 'metadata-freezing' stanza (reason 2 of 3): String: 10: Invalid value at 'max-interval': No number in duration value 'like'",
+          "Failed to parse Carboniter 'metadata-freezing' stanza (reason 3 of 3): String: 11: multiplier has type STRING rather than NUMBER"
```
Comment (cjllanwarne) [inline hybridCarboniteMetadataService/src/test/scala/cromwell/services/metadata/hybridcarbonite/HybridCarboniteConfigSpec.scala:192]: Honestly, I was hoping for more `"turtles"` in this error message

File: cromwell.example.backends/cromwell.examples.conf (hunk 1 above)
Comment (mcovarr) [reply cromwell.example.backends/cromwell.examples.conf:539]: fuzzy math

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,16 @@
 # Cromwell Change Log
 
+## 50 Release Notes
+
+### Changes and Warnings
+
+#### Metadata Archival Config Change
+
+Cromwell's metadata archival configuration has undergone a minor rework for consistency,
```
Comment (cjllanwarne) [inline CHANGELOG.md:9]:
```suggestion
**Note:** Unless you have already opted-in to GCS-archival of metadata during its development, this change will not affect you.
Cromwell's metadata archival configuration has changed in a backwards incompatible way to increase consistency,
```

File: hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridCarboniteConfig.scala [hunk 5]
```diff
@@ -40,35 +45,39 @@ case object InactiveMetadataDeletionConfig extends MetadataDeletionConfig
...
-        HybridCarboniteFreezeScanConfig().validNelCheck
-      }
+        val errorOrFreezingConfig: ErrorOr[MetadataFreezingConfig] = (initialInterval, maxInterval, multiplier, minimumSummaryEntryId, debugLogging) flatMapN {
+          case (i, m, x, s, d) =>
+            i match {
+              case f: FiniteDuration =>
+                for {
+                  _ <- if (s.exists(_ < 0)) "`metadata-freezing.minimum-summary-entry-id` must be greater than or equal to 0. Omit or set to 0 to allow all entries to be summarized.".invalidNel else "".validNel
+                  _ <- if (f > m) s"'max-interval' $m should be greater than or equal to finite 'initial-interval' $f".invalidNel else "".validNel
+                  _ <- if (x > 1) "".validNel else "`metadata-freezing.multiplier` must be greater than 1 in Carboniter 'metadata-freezing' stanza".invalidNel
```
Comment (cjllanwarne) [inline hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/HybridCarboniteConfig.scala:74]: TOL if these validation steps were done as part of the initial parsing, I think you could take advantage of the "all errors at the same time" property of the applicative `flatMapN`. This way, you'll only get one error at a time.

**Pull request merged**